onnxscript/ir/_external_data.py

```diff
@@ -6,11 +6,37 @@
 
 __all__ = ["set_base_dir"]
 
+import dataclasses
 import os
-from typing import Iterator
+from typing import Iterator, Sequence
 
 from onnxscript.ir import _core, _enums, _protocols, traversal
 
+# Note: If needed in future, add these as parameters to the function calls
+# align_offset: Offset will always be page aligned and alloction granularity aligned for mmap support. This is done by padding previous tensor data with zeros keeping same length. Tensor data will be aligned if > align_threshold
+_ALIGN_OFFSET = True
+# align_threshold: Alignment threshold for size of data. Having a low threshold will waste file space for small initializers. Only when tensor's data is > the page_align_threshold it will be force aligned.
+_ALIGN_THRESHOLD = 1048576  # 1MB
+# allocation_granularity: The allocation Granularity for mmap() support. Typically 64KB for Windows & 4KB for other OSes.
+_ALLOCATION_GRANULARITY = 65536  # 64KB
+
+
+@dataclasses.dataclass
+class _ExternalDataInfo:
+    """
+    A class that stores information about a tensor that is to be stored as external data.
+
+    Attributes:
+        name: The name of the tensor that is to be stored as external data.
+        offset: The offset is used to determine where exactly in the file the external data is written to.
+        length: Stores the size of the tensor.
+    """
+
+    def __init__(self, name: str | None, offset: int, length: int):
+        self.name = name
+        self.offset = offset
+        self.length = length
```

Comment:
Refer to https://docs.python.org/3/library/dataclasses.html for usage: we only need to list the attributes and their types